dotnet-grad/Controllers/UserController.cs

```diff
@@ -0,0 +1,49 @@
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Net;
+using System.Threading.Tasks;
+using dotnet_grad.Models;
+using Microsoft.AspNetCore.Mvc;
+
+namespace dotnet_grad.Controllers
+{
+  [ApiController]
+  [Route("api/[controller]")]
+  public class UserController : ControllerBase
+  {
+    TestContext testContext;
+    public UserController(TestContext _testContext)
+    {
+      testContext = _testContext;
+    }
+
+    [HttpGet("GetAllUsers")]
+    public List<UserModel> getUsers()
```

Comment:
ALWAYS have a response DTO of some sorts, NEVER return the Model/Entity as is perceived from the database. DTO's are contracts between you and your clients and allow you to add additional abstraction and introduce a level of backwards compatibility with existing integrations should your core/model change.